src/Controller/Admin/Asset/AssetHelperController.php

```diff
@@ -340,7 +339,16 @@ protected function getFieldGridConfig(array $field, string $language = '', strin
             $result['layout']['fieldtype'] = 'manyToOneRelation';
             $result['layout']['subtype'] = $type;
         }
-
+        $eventDispatcher = \Pimcore::getKernel()->getContainer()->get('event_dispatcher');
```

Comment:
Inject the dispatcher instead of getting it directly from the containier:
```
public function __construct(
        protected EventDispatcherInterface $eventDispatcher,
    ) {
    }
```